samples/mqtt5_pubsub.md

```diff
@@ -71,3 +71,122 @@ You can also pass a Certificate Authority file (CA) if your certificate and key
 # For Windows: replace 'python3' with 'python' and '/' with '\'
 python3 mqtt5_pubsub.py --endpoint <endpoint> --cert <file> --key <file> --ca_file <file>
 ```
+
+## **Alternate Connection Configuration Methods supported by AWS IoT Core**
+### **Optional Keyword Arguments**
+All lifecycle events and the callback for publishes received by the MQTT5 Client should be added to the builder on creation of the Client. A full list of accepted arguments can be found in the API guide.
+
+* [Direct MQTT with X509-based mutual TLS](#direct-mqtt-with-x509-based-mutual-tls)
+* [Direct MQTT with Custom Authentication](./mqtt5_custom_authorizer_connect.md)
+* [Direct MQTT with PKCS11 Method](./mqtt5_pkcs11_connect.md)
+* [Direct MQTT with PKCS12 Method](#direct-mqtt-with-pkcs12-method)
+* [MQTT over Websockets with Sigv4 authentication](#mqtt-over-websockets-with-sigv4-authentication)
+* [MQTT over Websockets with Cognito authentication](#mqtt-over-websockets-with-cognito-authentication)
+### HTTP Proxy
+* [Adding an HTTP Proxy](#adding-an-http-proxy)
+
+#### **Direct MQTT with X509-based mutual TLS**
+For X509 based mutual TLS, you can create a client where the certificate and private key are configured by path:
+
+```python
+    # X.509 based certificate file
+    certificate_file_path = "<certificate file path>"
+    # PKCS#1 or PKCS#8 PEM encoded private key file
+    private_key_filePath = "<private key file path>"
```

Comment:
Nit: Naming inconsistency between `certificate_file_path` and `private_key_filePath`.

Reply:
Below we use `*_filepath` for function parameter names, maybe we should stick to this.

Reply:
Good catch. It's supposed to be `cert_filepath` and `pri_key_filepath`. Changed here and in the MQTT5 Userguide!